src/sdk/namespace_fs.js

```diff
@@ -1390,35 +1398,89 @@ class NamespaceFS {
         }
     }
 
+    _get_part_data_path(params) {
+        return path.join(params.mpu_path, `parts-size-${params.size}`);
+    }
+
+    _get_part_md_path(params) {
+        return path.join(params.mpu_path, `part-${params.num}`);
+    }
+
+    // optimized version of upload_multipart - 
+    // 1. if size is pre known -
+    //    1.1. calc offset
+    //    1.2. upload data to by_size file in offset position
+    //    1.3. set on the part_md_file size, offset and etag xattr
+    // 2. else -
+    //    2.1. upload data to part_md_file
+    //    2.2. calc offset
+    //    2.3. copy the bytes to by_size file
+    //    2.4. set on the part_md_file size, offset and etag xattr
     async upload_multipart(params, object_sdk) {
-        // We can use 'wt' for open mode like we do for upload_object() when
-        // we figure out how to create multipart upload using temp files.
-        const open_mode = 'w';
+        let open_mode = 'r+';
+        const md_open_mode = 'w+';
```

Comment:
we can add a single open mode if that can help here -
```
    { "w*", O_CREAT | O_RDWR },
```

Reply:
instead, what I did it to use * as truncate remover, let me know what you think

Reply:
I'd summon @eshelmarc to comment on that. WDYT on this mode. We don't want to truncate, but keep writing to the file in offsets that are multiplies of the part size. Romy said that O_APPEND is also not suitable so we wanted to open with just O_CREAT - Does this make sense?